apps/client/src/features/operator/operator-event/OperatorEvent.tsx

```diff
@@ -0,0 +1,73 @@
+import { RefObject } from 'react';
+import { OntimeEvent, UserFields } from 'ontime-types';
+
+import DelayIndicator from '../../../common/components/delay-indicator/DelayIndicator';
+import { useTimer } from '../../../common/hooks/useSocket';
+import { cx, getAccessibleColour } from '../../../common/utils/styleUtils';
+import { formatTime } from '../../../common/utils/time';
+
+import style from './OperatorEvent.module.scss';
+
+interface OperatorEventProps {
+  data: OntimeEvent;
+  cue: string;
+  isSelected: boolean;
+  subscribed: keyof UserFields | null;
+  subscribedAlias: string;
+  showSeconds: boolean;
+  isPast: boolean;
+  selectedRef?: RefObject<HTMLDivElement>;
+}
+
+// extract this to contain re-renders
+function RollingTime() {
+  const timer = useTimer();
+  return <>{formatTime(timer.current, { showSeconds: true, format: 'hh:mm:ss' })}</>;
+}
+
+export default function OperatorEvent(props: OperatorEventProps) {
+  const { data, cue, isSelected, subscribed, subscribedAlias, showSeconds, isPast, selectedRef } = props;
+
+  const start = formatTime(data.timeStart, { showSeconds });
+  const end = formatTime(data.timeEnd, { showSeconds });
+
+  const cueColours = data.colour && getAccessibleColour(data.colour);
```

Comment:
I believe this `cueColours` variable type is `boolean | object`. Does that make sense when it's spread into the `styles` prop on line 50?

Reply:
That is correct. in the case `data.colour `is evaluated to falsy, `cueColours` will be a boolean with value false

I tried having a look on ECMA what the behaviour is for spreading booleans, but without luck on finding the right documentation

However I made a quick sandbox to find that spreading a falsy value always returned an empty object
` {...false}, {...undefined}, {...null}, {...""}, {...0}`

Thank you for the deep dive.
In this case, it seems like we are being graced with a language feature that helps us. The code could potentially be clearer, but I am unsure that the added complexity of the check would make it worth it

Reply:
Can't `getAccessibleColour` just return `{}` if `data.colour` is falsey?